*: remove use of cockroachdb/circuitbreaker package

erikgrinaker opened the pull request:
This package is no longer used. RPC circuit breakers use pkg/util/circuit instead.

Resolves #105890.
Touches #99191.
Epic: none
Release note: None

erikgrinaker committed:
rpc: remove Context.BreakerFactory

This is no longer used after the recent RPC circuit breaker refactor.

Epic: none
Release note: None

erikgrinaker committed:
grpcutil: add `RequestDidNotStart` case for circuit breaker errors

Epic: none
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/110892)
<!-- Reviewable:end -->

erikgrinaker committed:
*: remove use of cockroachdb/circuitbreaker package

This package is no longer used. RPC circuit breakers use
pkg/util/circuit instead.

Epic: none
Release note: None

erikgrinaker commented:
TFTR!

bors r+
